src/java.base/share/classes/java/util/stream/ForEachOps.java

```diff
@@ -499,7 +511,8 @@ else if (spliterator != null) {
             // "happens-before" completion of the associated left-most leaf task
             // of right subtree (if any, which can be this task's right sibling)
             //
-            ForEachOrderedTask<S, T> leftDescendant = completionMap.remove(this);
+            @SuppressWarnings("unchecked")
+            var leftDescendant = (ForEachOrderedTask<S, T>)NEXT.getAndSet(this, null);
```

Comment:
I don’t think that this needs a `@SuppressWarnings("unchecked")`, as casts of signature polymorphic return values don’t emit unchecked warnings (unless that changed recently):
```suggestion
            var leftDescendant = (ForEachOrderedTask<S, T>) NEXT.getAndSet(this, (ForEachOrderedTask<S, T>) null);
```

Reply:
Thanks for having a look at the PR, you're right, the suppression isn't needed.